lexer.go

```diff
@@ -97,9 +103,39 @@ func (s *sshLexer) lexRvalue() sshLexStateFn {
 			s.skip()
 			return s.lexVoid
 		case '#':
+			if !isSingleQuoted && !isDoubleQuoted {
+				s.emitWithValue(tokenString, growingString)
+				s.skip()
+				return s.lexComment(s.lexVoid)
+			}
+		case eof:
+			s.next()
+		}
+		if next == eof {
+			break
+		}
+		growingString += string(next)
+		s.next()
+	}
+	s.emit(tokenEOF)
+	return nil
+}
+
+func (s *sshLexer) lexRReservedvalue() sshLexStateFn {
```

Comment:
I think there is no need to add a new function, just add an argument, such as:
```go
func (s *sshLexer) lexRvalue(noComment bool) sshLexStateFn {
  for {
    switch next {
    case '#':
      if !noComment {
        // ...
      }
    }
  }
}
```

Reply:
I change `noComment `  to  `ignoreHash` for more human readable.

Reply:
> I change `noComment ` to `ignoreHash` for more human readable.

Is `notEndWithComment` better?